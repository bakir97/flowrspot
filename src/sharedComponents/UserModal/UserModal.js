import React, { Component } from "react";
import Modal from "../Modal";
import styles from "./UserModal.module.scss";
import SignInButton from "../SignInButton";
import Loader from "../Loader";
import Profile from "./Profile";
import SignInForm from "./SignInForm";
export default class CreateAccount extends Component {
  handleOkButton = () => {
    const { openModal, closeModal, modalType } = this.props;
    if (modalType === "login") {
      window.document.body.style.overflow = "auto";
      return closeModal();
    }
    return openModal("login");
  };
  handleProfileButton = () => {
    const { openModal } = this.props;

    return openModal("profile");
  };
  render() {
    const {
      loading,
      isUserSuccess,
      modalType,
      user,
      logout,
      closeModal
    } = this.props;
    const isLogin = modalType === "login" ? true : false;

    if (modalType === "profile") {
      return <Profile user={user} logout={logout} closeModal={closeModal} />;
    }
    if (isUserSuccess) {
      return (
        <Modal>
          <div className={styles.success_container}>
            <p className={styles.success_text}>
              {isLogin
                ? "Congratulations! You have successfully logged into FlowrSpot!"
                : "Congratulations! You have successfully signed up for FlowrSpot!"}
            </p>
            <SignInButton
              title="OK"
              type="submit"
              onClickFunction={this.handleOkButton}
              buttonStyle={isLogin ? "button_margin" : false}
            />
            {isLogin && (
              <SignInButton
                title="Profile"
                onClickFunction={this.handleProfileButton}
              />
            )}
          </div>
        </Modal>
      );
    }
    if (loading) {
      return (
        <Modal
          modalText={
            isLogin ? "I don’t want to Login" : "I don’t want to register"
          }
        >
          <Loader />
        </Modal>
      );
    }
    return <SignInForm {...this.props} />;
  }
}
