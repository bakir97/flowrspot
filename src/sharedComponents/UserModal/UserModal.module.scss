.success_container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  .success_text {
    color: black;
    text-align: center;
  }
}
