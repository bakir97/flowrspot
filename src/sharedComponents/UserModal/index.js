import UserModal from "./UserModal";
import { connect } from "react-redux";
import { registerUser, loginUser, logout } from "../../redux/user/actions";
import {
  isUserLoading,
  isUserSuccess,
  getErrors,
  getUser
} from "../../redux/user/selectors";
import { getModalType } from "../../redux/modal/selectors";
import { openModal, closeModal } from "../../redux/modal/actions";
import { errorUser } from "../../redux/user/actions";
const mapStateToProps = state => ({
  loading: isUserLoading(state),
  isUserSuccess: isUserSuccess(state),
  modalType: getModalType(state),
  errors: getErrors(state),
  user: getUser(state)
});

const mapDispatchToProps = {
  registerUser,
  openModal,
  errorUser,
  loginUser,
  closeModal,
  logout
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserModal);
