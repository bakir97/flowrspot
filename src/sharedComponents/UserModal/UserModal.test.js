import React from "react";
import { shallow } from "enzyme";
import UserModal from "./UserModal";
describe("UserModal", () => {
  it("should render correctly", () => {
    const component = shallow(<UserModal modalType />);
    expect(component).toMatchSnapshot();
  });
  it("should render Profile", () => {
    const component = shallow(<UserModal modalType="profile" />);
    expect(component).toMatchSnapshot();
  });
  it("should render SignInForm", () => {
    const component = shallow(<UserModal modalType />);
    expect(component).toMatchSnapshot();
  });
  it("should render Loading Component", () => {
    const component = shallow(<UserModal modalType loading />);
    expect(component).toMatchSnapshot();
  });

  it("should render Success Component ", () => {
    const component = shallow(<UserModal modalType isUserSuccess />);
    expect(component).toMatchSnapshot();
  });
});
