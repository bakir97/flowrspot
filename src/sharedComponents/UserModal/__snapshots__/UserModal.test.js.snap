// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModal should render Loading Component 1`] = `
<Connect(Modal)
  modalText="I donâ€™t want to register"
>
  <Loader />
</Connect(Modal)>
`;

exports[`UserModal should render Profile 1`] = `<Profile />`;

exports[`UserModal should render SignInForm 1`] = `
<SignInForm
  modalType={true}
/>
`;

exports[`UserModal should render Success Component  1`] = `
<Connect(Modal)>
  <div
    className="success_container"
  >
    <p
      className="success_text"
    >
      Congratulations! You have successfully signed up for FlowrSpot!
    </p>
    <SignInButton
      buttonStyle={false}
      onClickFunction={[Function]}
      title="OK"
      type="submit"
    />
  </div>
</Connect(Modal)>
`;

exports[`UserModal should render correctly 1`] = `
<SignInForm
  modalType={true}
/>
`;
